### PR TITLE
MRD-270_write-recommendations-to-database: Added endpoint to persist lightweight recommendation.

### PR DESCRIPTION
**Pull request by bills7281**
I've kept the recommendation response/request quite lightweight here as -

There are a few outstanding questions on what it will look like, for example -
questions -
* What is inside 'offence', is this just main offence? Or all under active convictions?
* Should the user supply the offender's name, offences etc in the request? Or just the CRN which can then use Delius/ARN to fetch all that information?

[[](url)](https://miro.com/app/board/uXjVOuO1zFk=/?moveToWidget=3458764527484200310&cot=14)

This PR contains a Spring Data JPA repo and entity, these two classes can be swapped out for their nosql counterparts in a subsequent task.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/ContactHistoryService.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecis
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.ContactSummaryResponse
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.ndelius.ReleaseSummaryResponse
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.reader.ContactGroupsCsvReader
+import kotlin.streams.toList
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/ContactHistoryService.kt:12]: Do we need these imports? There are no other changes to that file

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/LicenceConditionsService.kt [hunk 2]
```diff
@@ -8,6 +8,7 @@ import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecis
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.LicenceConditionsResponse
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.Offence
 import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.ndelius.ReleaseSummaryResponse
+import kotlin.streams.toList
```
Comment (jackmaloney) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/LicenceConditionsService.kt:12]: Do we need this import? There are no other changes to that file

File: src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/PersonDetailsControllerTest.kt [hunk 3]
```diff
@@ -16,7 +16,7 @@ class PersonDetailsControllerTest(
 ) : IntegrationTestBase() {
 
   @Test
-  fun `retrieves person details`() {
+  fun `retrieves recommendation`() {
```
Comment (jackmaloney) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/PersonDetailsControllerTest.kt:19]: Is this test name supposed to be changed?

Comment (bills7281) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/integration/controller/PersonDetailsControllerTest.kt]: Corrected!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/LicenceConditionsService.kt (hunk 2 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/LicenceConditionsService.kt:12]: My IDE complains and errors when it is not there :(

File: src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/ContactHistoryService.kt (hunk 1 above)
Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/ContactHistoryService.kt:12]: My IDE complains and errors when it is not there :(

Comment (bills7281) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/ContactHistoryService.kt]: Removed

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**